src/Model/Subsite.php

```diff
@@ -652,78 +651,83 @@ public function canEdit($member = false)
      */
     public function getCMSFields()
     {
-        if ($this->ID != 0) {
-            $domainTable = new GridField(
-                'Domains',
-                _t(__CLASS__ . '.DomainsListTitle', 'Domains'),
-                $this->Domains(),
-                GridFieldConfig_RecordEditor::create(10)
-            );
-        } else {
-            $domainTable = new LiteralField(
-                'Domains',
-                '<p>' . _t(
-                    __CLASS__ . '.DOMAINSAVEFIRST',
-                    'You can only add domains after saving for the first time'
-                ) . '</p>'
-            );
-        }
+        $this->beforeUpdateCMSFields(function (FieldList $fields) {
+            if ($this->exists()) {
+                // Add a GridField for domains to a new tab if the subsite has already been created
+                $fields->addFieldsToTab('Root.Domains', [
+                    GridField::create(
+                        'Domains',
+                        _t(__CLASS__ . '.DomainsListTitle', 'Domains'),
+                        $this->Domains(),
+                        GridFieldConfig_RecordEditor::create(10)
+                    )
+                ]);
+            }
 
-        $languageSelector = new DropdownField(
-            'Language',
-            $this->fieldLabel('Language'),
-            Injector::inst()->get(IntlLocales::class)->getLocales()
-        );
+            // Remove the default scaffolded blacklist field, we replace it with a checkbox set field
+            // in a wrapper further down. The RedirectURL field is currently not in use.
+            $fields->removeByName(['PageTypeBlacklist', 'RedirectURL']);
+
+            // Add the heading to the top of the fields list
+            $fields->fieldByName('Root.Main')
+                ->unshift(HeaderField::create('ConfigForSubsiteHeaderField', 'Subsite Configuration'));
```

Comment:
Header label not translatable. Should it be?

Reply:
Yep, thanks